app/src/main/java/com/example/tasky/feature_authentication/domain/use_case/Login.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.tasky.feature_authentication.domain.use_case
+
+import com.example.tasky.feature_authentication.data.util.AuthResult
+import com.example.tasky.feature_authentication.domain.repository.AuthRepository
+import com.example.tasky.feature_authentication.domain.util.UseCaseResult
+import com.example.tasky.feature_authentication.domain.validation.UserDataValidator
+
+class Login(
+    private val repository: AuthRepository,
+    private val userDataValidator: UserDataValidator
+) {
```

Comment:
This for example is a great use case, since it contains actual logic which can now be perfectly reused 👍